src/main/java/aeneas/views/LightningWidgetView.java

```diff
@@ -35,8 +55,29 @@ public LightningWidgetView(LightningLevel levelModel){
     button.setUserData(this);
   }
 
+  public Level resetLevelModel(Level level) {
+    if (level instanceof LightningLevel) {
+      this.level = (LightningLevel)level;
+      return level;
+    }
+    this.level = new LightningLevel(level);
+    timeSelect.valueProperty().removeListener(controller);
+    timeSelect.getValueFactory().setValue(this.level.getAllowedTime());
+    controller = new SetTimeController();
+    timeSelect.valueProperty().addListener(controller);
+    return this.level;
+  }
+
   @Override
   public RadioButton getButton() {
     return LightningWidgetView.button;
   }
+
+  @Override
+  public void refresh() {
+    timeSelect.valueProperty().removeListener(controller);
+    timeSelect.getValueFactory().setValue(level.getAllowedTime());
+    timeSelect.valueProperty().addListener(controller);
```

Comment:
I know why we're doing this but I just want to officially proclaim it's super hacky and I don't like it.